packages/asc-ui/src/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useImperativeHandle } from 'react'
+import useActionOnEscape from '../../utils/hooks/useActionOnEscape'
+import useFocusOnRender from '../../utils/hooks/useFocusOnRender'
+import TextAreaStyle from './TextAreaStyle'
+
+export interface TextAreaMethods
+  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {
+  setTextAreaRef?: (ref: React.RefObject<HTMLTextAreaElement>) => void
+}
+
+export interface TextAreaProps extends TextAreaMethods {
+  keepFocus?: boolean
+  blurOnEscape?: boolean
+  focusOnRender?: boolean
+  error?: string
+}
+
+const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
+  ({ blurOnEscape, focusOnRender, value, error, ...props }, externalRef) => {
+    const { onKeyDown } = props
+    const TextAreaRef = React.useRef<HTMLTextAreaElement>(null)
+
+    useImperativeHandle(
+      externalRef,
+      () => TextAreaRef.current as HTMLTextAreaElement,
+    )
+    useFocusOnRender(TextAreaRef, focusOnRender)
+
+    const { onKeyDown: onKeyDownHook } = useActionOnEscape(() => {
+      if (TextAreaRef.current) {
+        TextAreaRef.current.blur()
+      }
+    })
+
+    const handleOnKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+      if (onKeyDown) {
+        onKeyDown(e)
+      }
+      if (blurOnEscape) {
+        onKeyDownHook(e)
+      }
+    }
+
+    return (
+      <TextAreaStyle
+        ref={TextAreaRef}
+        {...props}
```

Comment:
Make sure that the props are spread at the end, othewise they might be overwritten at a later point. It's also conventional to call this variable `otherProps` since it contains the props that are not explicitly used in the component.

Reply:
Done